added README and Json postman (#14)
keda-vlad/bookstore: README.md

# Bookstore API: Enhancing Your Bookstore Experience

Welcome to the Bookstore API project! Our powerful and secure application, developed using Java Spring Boot, aims to enrich online book purchases for all book enthusiasts. We have combined various features and advanced technologies to provide users with a secure, comprehensive and convenient e-commerce platform. Below, we will introduce you to the key aspects and functional capabilities of our application.

## Features:

### Security:

> In our application, we use JWT tokens to ensure secure authentication and authorization. This guarantees that only verified users can access sensitive features.

Reviewer: ```suggestion
Bookstore uses JWT tokens to ensure secure authentication and authorization. This guarantees that only verified users can access sensitive features.
```

- Spring Security: Ensuring application and user data security.
- Spring MVC: Developing a web application using the Model-View-Controller (MVC) pattern for better code organization and separation of business logic, views, and control.
- MapStruct: Efficient mapping of models to Data Transfer Objects (DTOs).
- Docker: Containerization for easy deployment and scalability.
- OpenAPI: Documenting our RESTful API to provide clear information about available resources, requests, and response formats.
- Liquibase: Managing and versioning the database schema for easy schema changes and data schema consistency in different environments.

## Functionality:

In our API, functionality is divided for Users and Admin:

| Controller           | User                                                     | Admin                      |
|-----------------------|----------------------------------------------------------|------------------------------------|
| BookController        | Book search with filtering, pagination and specification | Read, update, delete functionality |
| CategoryController    | Category and books by categories search                  | Read, update, delete functionality |

> | AuthenticationController | Registration, authentication                             | Ability to change user roles |

Reviewer: Please align table

Author: It's align in preview mode
![image](https://github.com/keda-vlad/bookstore/assets/135035436/0c5d677d-d1c7-4548-b2a9-a6c1c6edc207)

   Via the console:
    - Execute the command: `git clone https://github.com/keda-vlad/bookstore.git`
    - Run the following commands:
        - `mvn clean install`
        - `docker build -t your_image_name`
        - `docker-compose up --build`

## Usage
    - Postman:
        - Open Postman.
        - Import the [file](book-store.postman_collection.json) with requests.

## Contribution

   ### Issues and Solutions

   **Issue 1: N+1 Problem**
    - Solution: Use `@Query(join fetch)` or `@EntityGraph` annotations for methods with specifications.

   **Issue 2: Authentication Security**
    - Solution: We implemented Spring Security and JWT tokens for robust protection.

   **Issue 3: Exception Handling**
    - Solution: We developed a global exception handler and specific exceptions to improve error reporting.

# Conclusion:

   The Bookstore API provides a robust foundation for building a book-selling platform. We welcome contributions from the community to enhance the Bookstore project. Whether you want to fix a bug, improve an existing feature, or propose a new one, your contributions are valuable to us.

   ### Upcoming Expected Updates:

>    We plan to add user recommendation functionality.

Reviewer: Who we? I'm alone here

## Contact Us:

>    For any inquiries or suggestions, please email us at vladyslav.keda@gmail.com.

Reviewer: Please record loom video
